enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -500,6 +527,12 @@ func (np *nodePool) NodeCount(ctx context.Context, taskSize *scpb.TaskSize) (int
 	if len(np.nodes) == 0 {
 		return 0, status.UnavailableErrorf("No registered executors in pool %q with os %q with arch %q.", np.key.pool, np.key.os, np.key.arch)
 	}
+	// If the task is unsized, any node can execute the task, since we cap
+	// unsized task limits so that they can always fit on any registered
+	// executor.
```

Comment:
this may not always be the case right? imagine if we had a bunch of executors of size 1 and size 10, we might want to run it at size 10 and see how much it takes, not assume it's max size of 1?